src/test/java/UnitTests/JsonParserTests/JsonParserReadTests.java

```diff
@@ -0,0 +1,76 @@
+package UnitTests.JsonParserTests;
+
+import com.google.gson.JsonSyntaxException;
+import org.junit.gen5.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.testng.Assert;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import parser.Parser;
+import shop.Cart;
+import java.io.File;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class JsonParserReadTests {
+
+    @ParameterizedTest
+    @ValueSource(strings = { "sample-cart.json" })
+    void readFromFileTest(String fileName) {
+        Parser parser = new JsonParser();
+        Cart cart;
+
+        cart = parser.readFromFile(new File("./src/test/java/UnitTests/TestData/" + fileName));
+        cart.showItems();
+
+        Assert.assertNotNull(cart);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "empty-items-cart.json" })
+    void readFromEmptyItemsTest(String fileName) {
+        Assertions.assertThrows(NullPointerException.class, () -> {
+            new JsonParser().readFromFile(new File("./src/test/java/UnitTests/TestData/" + fileName)).showItems();
+        });
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "wrong-datatype-cart.json" })
+    void readFromWrongDataTypeTest(String fileName) {
+        Assertions.assertThrows(NumberFormatException.class, () -> {
+            new JsonParser().readFromFile(new File("./src/test/java/UnitTests/TestData/" + fileName)).showItems();
+        });
+    }
```

Comment:
please take a look these both tests and readFromWrongExtensionTypeTest absolutely the same except a params value. No need to copy paste these tests, junit allows you to unite them in 1 test with a collection of params

Reply:
not quite the same. Each test catch different exception NullPointerException and NumberFormatException. I didnt find a way to be able to catch different exceptions in the same test. Only like that